Migrate library to .NET Standard

Jimmyson opened the pull request:
Looking to utilize the library in a .NET Core application, and thought it would be useful to have this migrated, supported and published to NuGet.

Namespaces have been preserved, and dependent libraries have been updated.
Unit tests still pass.
Removed Example WinForms application as no code remained.

Jimmyson committed:
Revert "Migrate Libraries to .NET Standard and Core"

This reverts commit b2229cb78cfb23bbe53d9cb8fc5610ae7223e367.

Signed-off-by: Jimmyson <jimmyson.2007@gmail.com>

Jimmyson committed:
Migrate Libraries to .NET Standard and Core

Jimmyson committed:
Remove Fingerprint Example Application

wo80 commented:
Thanks, I fixed the appveyor test runner.

Two remarks:

You made a lot of previously internal classes public. Is this because of the tests or is there any other reason? If it's because of the tests, you could add
```c#
[assembly: InternalsVisibleTo("AcoustID.Tests")] // or AcoustID.Core.Tests
```
to one of the source files and revert the changes (public -> internal).

Additionally, I don't like the `Core` suffix too much. Any good reason to add it to the project file names?

wo80 commented on src/AcoustID.Core/AcoustID.Core.csproj:
```diff
@@ -0,0 +1,7 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
```
Please add a property group containing the assembly information:
```xml
  <PropertyGroup>
    <GeneratePackageOnBuild>false</GeneratePackageOnBuild>
    <GenerateDocumentationFile>true</GenerateDocumentationFile>
    <PackageId>AcoustID.NET</PackageId>
    <Description>AcoustID fingerprinter and webservice access.</Description>
    <Product>AcoustID</Product>
    <Copyright>Copyright © Christian Woltering 2016-2019</Copyright>
    <Authors>Christian Woltering</Authors>
    <PackageProjectUrl>https://github.com/wo80/AcoustID.NET</PackageProjectUrl>
    <RepositoryUrl>https://github.com/wo80/AcoustID.NET</RepositoryUrl>
    <AssemblyVersion>1.3.3.0</AssemblyVersion>
    <FileVersion>1.3.3.0</FileVersion>
    <PackageTags>music audio fingerprint acoustid chromaprint</PackageTags>
    <Version>1.3.3</Version>
    <AssemblyName>AcoustID</AssemblyName>
    <RootNamespace>AcoustID</RootNamespace>
    <Configurations>Debug;Release</Configurations>
    <PackageLicenseUrl>https://github.com/acoustid/chromaprint/blob/master/LICENSE.md</PackageLicenseUrl>
  </PropertyGroup>
```

Jimmyson replied:
Changed have been implemented after the revert. See `AcoustID.csproj` in 085088c

Jimmyson commented:
My reason for adding Core was to indicate the change to Core framework, it was only in name though and I can revert this.

In regards to opening up the internal classes, I had access issues. I did this some time ago, and did not know about the assembly directive.

I can reattempt this to keep in line with access structure...

wo80 commented:
If you need access to those classes for your work, I don't mind making them public. I was just wondering ...

But I'd prefer if you remove `Core` from directory and project file names. Thanks!

Jimmyson commented:
I have taken a branch of the original fork, and followed through with checking the to see if's possible to expose the Internal classes. And the tests successfully pass.

![image](https://user-images.githubusercontent.com/8276052/68852896-eaf37280-072c-11ea-887e-cc753d75223c.png)

There is an attribute that can be added to the primary library's CSPROJ file that allows the "InternalVisableTo" flag
```xml
<ItemGroup>
    <AssemblyAttribute Include="System.Runtime.CompilerServices.InternalsVisibleTo">
        <_Parameter1>$(AssemblyName).Tests</_Parameter1>
    </AssemblyAttribute>
</ItemGroup>
```
https://blog.sanderaernouts.com/make-internals-visible-with-new-csproj-format

I can revert the file back to their original state, but since I changed the folder name to include the `Core` value. I will need to rollback the branch. Apologies for that.

Jimmyson committed:
Migrate to .Net Standard and Core
Preserve Assembly Name
Only Expose Internals to Test Library
Remove old AssemblyInfo and package defintion

Jimmyson commented:
I have rolled back my changes and reattempted the migration.
Preserving the Internals and removing the `Core` name from folders too.
I wasn't too sure about if you were also concerned about the `test` and `src` folders, so reverted that too.

But I have now implemented the defined changes you have requested for `Internals` and the property group

Jimmyson committed:
Remove Fingerprinter from Solution

wo80 commented:
Ok, this looks much cleaner. I'll merge and publish a new package this weekend.
